sparse_dot_topn/awesome_cossim_topn.py

```diff
@@ -39,12 +39,21 @@ def awesome_cossim_topn(A, B, ntop, lower_bound=0, use_threads=False, n_jobs=1):
 
     if K1 != K2:
         err_str = 'A matrix multiplication will be operated. A.shape[1] must be equal to B.shape[0]!'
-        raise ValueError(err_str)        
+        raise ValueError(err_str)
 
     idx_dtype = np.int32
 
     nnz_max = M*ntop
 
+    # basic check. if A or B are all zeros matrix, return all zero matrix directly
+    if len(A.indices) == 0 or len(A.data) == 0 or len(A.indptr) == 0 or \
```

Comment:
Since they are csr matrix, if len(A.indices) is zero, rest will be zero too.

Reply:
Done.

Reply:
Fixed. I also squashed the commits.